### PR TITLE
MINOR: Move upgraded docs from site to Kafka docs

### PR DESCRIPTION
**Pull request by bbejeck**
For the 2.7 release, we need to migrate some docs changes that went to `kafka-site` but didn't go into `kafka/docs`

This PR covers the `documentation.html` and `upgrade.html` changes.  Once these are merged to trunk, I'll cherry-pick them to the 2.7 branch

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
Note that most of this PR already exists in 2.6 and doesn't need a close look

File: docs/upgrade.html [hunk 1]
```diff
@@ -93,8 +135,49 @@ <h5><a id="upgrade_270_notable" href="#upgrade_270_notable">Notable changes in 2
...
+    <li> If you have overridden the message format version as instructed above, then you need to do one more rolling restart to
+        upgrade it to its latest version. Once all (or most) consumers have been upgraded to 0.11.0 or later,
+        change log.message.format.version to 2.6 on each broker and restart them one by one. Note that the older Scala clients,
+        which are no longer maintained, do not support the message format introduced in 0.11, so to avoid conversion costs
+        (or to take advantage of <a href="#upgrade_11_exactly_once_semantics">exactly once semantics</a>),
+        the newer Java clients must be used.
+    </li>
+</ol>
+
+<h5 class="anchor-heading"><a id="upgrade_260_notable" class="anchor-link"></a><a href="#upgrade_260_notable">Notable changes in 2.6.0</a></h5>
```
Comment (bbejeck) [inline docs/upgrade.html:180]: Current 2.6 docs

File: docs/upgrade.html [hunk 2]
```diff
@@ -114,17 +197,51 @@ <h5><a id="upgrade_260_notable" href="#upgrade_260_notable">Notable changes in 2
...
+    <li> If you have overridden the message format version as instructed above, then you need to do one more rolling restart to
+        upgrade it to its latest version. Once all (or most) consumers have been upgraded to 0.11.0 or later,
+        change log.message.format.version to 2.5 on each broker and restart them one by one. Note that the older Scala clients,
+        which are no longer maintained, do not support the message format introduced in 0.11, so to avoid conversion costs
+        (or to take advantage of <a href="#upgrade_11_exactly_once_semantics">exactly once semantics</a>),
+        the newer Java clients must be used.
+    </li>
+</ol>
+
+<h5 class="anchor-heading"><a id="upgrade_250_notable" class="anchor-link"></a><a href="#upgrade_250_notable">Notable changes in 2.5.0</a></h5>
```
Comment (bbejeck) [inline docs/upgrade.html:244]: current 2.6 docs as well

File: docs/upgrade.html [hunk 3]
```diff
@@ -21,6 +21,48 @@
 
 <h5><a id="upgrade_280_notable" href="#upgrade_280_notable">Notable changes in 2.8.0</a></h5>
 
+<h4><a id="upgrade_2_7_0" href="#upgrade_2_7_0">Upgrading to 2.7.0 from any version 0.8.x through 2.6.x</a></h4>
```
Comment (bbejeck) [inline docs/upgrade.html:24]: This whole section is new, but should be standard instructions

File: docs/documentation.html [hunk 4]
```diff
@@ -22,51 +22,56 @@
...
+    <h2 class="anchor-heading"><a id="operations" class="anchor-link"></a><a href="#operations">6. Operations</a></h2>
 
     <!--#include virtual="ops.html" -->
 
-    <h2><a id="security" href="#security">7. Security</a></h2>
+    <h2 class="anchor-heading"><a id="security" class="anchor-link"></a><a href="#security">7. Security</a></h2>
     <!--#include virtual="security.html" -->
 
-    <h2><a id="connect" href="#connect">8. Kafka Connect</a></h2>
+    <h2 class="anchor-heading"><a id="connect" class="anchor-link"></a><a href="#connect">8. Kafka Connect</a></h2>
```
Comment (bbejeck) [inline docs/documentation.html:74]: Current 2.6 docs part of AK site update

File: docs/documentation.html [hunk 5]
```diff
@@ -22,51 +22,56 @@
...
+ <!--#include virtual="../includes/_nav.htm" -->
+  <div class="toc-handle-container">
+    <div class="toc-handle">&lt;</div>
+  </div>
+  <div class="docs-nav">
+    <!--#include virtual="toc.html" -->
+  </div>
+  <div class="right">
+    <!--//#include virtual="../includes/_docs_banner.htm" -->
+
```
Comment (bbejeck) [inline docs/documentation.html:34]: Current docs, part of AK site upgrade

**Review by mjsax — APPROVED** [review]

**Pull request merged**